### PR TITLE
Support AWS CodeArtifact publishing location

### PR DESCRIPTION
**Pull request by zak-dd**

### CODE REVIEW

File: src/sbt-test/sbt-config/scalafmt/build.sbt [hunk 1]
```diff
@@ -6,8 +6,8 @@ lazy val root = (project in file("."))
     scalaVersion := "2.12.12",
     // Default publishArtifact is false
     publishLocal := {
-      val x = publishArtifact.value
-      assert(x == false, s"publishArtifact should be false: $x")
+      val x = (publish / skip).value
```
Comment (puchta) [inline src/sbt-test/sbt-config/scalafmt/build.sbt:9]: please adjust (or delete) the comment in the line 7

File: src/sbt-test/sbt-config/scalastyle-disabled/build.sbt [hunk 2]
```diff
@@ -6,8 +6,8 @@ lazy val root = (project in file("."))
     scalaVersion := "2.12.12",
     // Default publishArtifact is false
     publishLocal := {
-      val x = publishArtifact.value
-      assert(x == false, s"publishArtifact should be false: $x")
+      val x = (publish / skip).value
```
Comment (puchta) [inline src/sbt-test/sbt-config/scalastyle-disabled/build.sbt:9]: please adjust (or delete) the comment in the line 7

File: src/sbt-test/sbt-config/publishing-enabled/build.sbt [hunk 3]
```diff
@@ -8,6 +8,8 @@ lazy val root = (project in file("."))
     publishLocal := {
       val x = publishArtifact.value
       assert(x == true, s"publishArtifact should be true: $x")
+      val skipPublish = (publish / skip).value
```
Comment (puchta) [inline src/sbt-test/sbt-config/publishing-enabled/build.sbt:11]: please adjust (or delete) the comment in the line 7

File: src/sbt-test/sbt-config/default/build.sbt [hunk 4]
```diff
@@ -6,8 +6,8 @@ lazy val root = (project in file("."))
     scalaVersion := "2.12.12",
     // Default publishArtifact is false
     publishLocal := {
-      val x = publishArtifact.value
-      assert(x == false, s"publishArtifact should be false: $x")
+      val x = (publish / skip).value
```
Comment (puchta) [inline src/sbt-test/sbt-config/default/build.sbt:9]: please adjust (or delete) the comment in the line 7

File: src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala [hunk 5]
```diff
@@ -340,12 +346,20 @@ object SbtConfigPlugin extends AutoPlugin {
...
+                codeArtifactToken := obtainCodeArtifactToken(awsAccountID, region, domain, awsCredentials),
+                credentials += Credentials(s"${domain}/${repoName}", s"${domain}-${awsAccountID}.d.codeartifact.${region}.amazonaws.com", "aws", codeArtifactToken.value)
+              )
           }
         }
       } else {
         universalCommonPublishingSettings ++ Seq(
           organization := "com.dotdata",
-          publishArtifact := false
+          publish / skip := true
```
Comment (dot-vlad) [inline src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala:362]: Is that ok to do it in a separate PR to make sure this PR doesn't cause any regression?

**Review by dot-vlad — COMMENTED** [review]
Two comments

File: src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala (hunk 5 above)
Comment (zak-dd) [reply src/main/scala/com/dotdata/sbt/SbtConfigPlugin.scala:362]: Ok

File: src/main/scala/com/dotdata/sbt/SbtConfigKeys.scala [hunk 6]
```diff
@@ -1,5 +1,12 @@
 package com.dotdata.sbt
 
-trait SbtConfigKeys
+import sbt.taskKey
+
+trait SbtConfigKeys {
+  val codeArtifactToken = taskKey[String](
```
[the remark this answers is not in the document]
Comment (zak-dd) [reply src/main/scala/com/dotdata/sbt/SbtConfigKeys.scala:6]: It isn't needed. I found it useful at some point for debugging, but I think it isn't valuable anymore. I'll remove it

**Review by puchta — APPROVED** [review]
I'm not sure what's going on here. But looks legit.

File: src/sbt-test/sbt-config/scalafmt/build.sbt (hunk 1 above)
Comment (zak-dd) [reply src/sbt-test/sbt-config/scalafmt/build.sbt:9]: Thanks. I'm reverting this change and I'll introduce it in the next PR based on @dot-vlad's suggestion: https://github.com/ramencloud/sbt-config/pull/49#discussion_r740024930 I'll apply this comment in the next PR then

**Review by dot-vlad — APPROVED** [review]
